emhttp/plugins/dynamix/include/NetworkInfo.php

```diff
@@ -24,19 +24,18 @@ function port($eth) {
   $sys = "/sys/class/net";
   if (substr($eth,0,4)=='wlan') return $eth;
   $x = preg_replace('/[^0-9]/','',$eth);
-  return file_exists("$sys/br{$x}") ? "br${x}" : (file_exists("$sys/bond{$x}") ? "bond{$x}" : "eth{$x}");
+  return file_exists("$sys/br{$x}") ? "br{$x}" : (file_exists("$sys/bond{$x}") ? "bond{$x}" : "eth{$x}");
 }
 
 exec("grep -Po 'nameserver \K\S+' /etc/resolv.conf 2>/dev/null",$ns);
-$eth    = $_POST['port'];
-$vlan   = $_POST['vlan'];
+$eth    = $_POST['port'] ?? '';
```

Comment:
Can we add some validation on anything that comes from `$POST`?   like `$_POST['port']` has to be numeric

Reply:
I added a default number to the 'port' function in case the variable is rubbish (it should not)
The 'port' function always returns a valid name regardless of input